app/fs/fs.go

```diff
@@ -185,34 +186,69 @@ func Unzip(src, dest string) error {
 	return nil
 }
 
-// Ungzip uncompresses the given tar.gz file
-func Ungzip(gzStream, dest string) error {
+// Untar takes a destination path and a reader; a tar reader loops over the tarfile
+// creating the file structure at 'dst' along the way, and writing any files
+func Untar(gzStream string, dst string) error {
+	r, err := os.Open(gzStream)
+	gzr, err := gzip.NewReader(r)
 
-	log.Println("Ungzip the tar.gz from " + dest)
-
-	gzReader, err := os.Open(gzStream)
+	defer gzr.Close()
 	if err != nil {
 		return err
 	}
-	defer gzReader.Close()
 
-	gzArchive, err := gzip.NewReader(gzReader)
-	if err != nil {
-		return err
-	}
-	defer gzArchive.Close()
+	tr := tar.NewReader(gzr)
 
-	dest = filepath.Join(dest, gzArchive.Name)
+	for {
+		header, err := tr.Next()
 
-	writer, err := os.Create(dest)
-	if err != nil {
-		return err
-	}
-	defer writer.Close()
+		switch {
+
+		// if no more files are found return
+		case err == io.EOF:
+			return nil
+
+			// return any other error
```

Comment:
are these supposed to be indented so much? makes it look like they refer to the stuff from lines 207-208